qiita_db/study.py

```diff
@@ -673,7 +674,13 @@ def sample_template(self):
         -------
         SampleTemplate id
         """
-        return self._id
+        with TRN:
+            sql = """SELECT EXISTS(SELECT *
+                                   FROM qiita.study_sample
+                                   WHERE study_id = %s)"""
+            TRN.add(sql, [self.id])
+            exists = TRN.execute_fetchlast()
+        return self._id if exists else None
```

Comment:
I would prefer raising an error, but that may be rather draconian for a
getter.

On (Oct-05-15|15:41), Antonio Gonzalez wrote:
> > @@ -673,7 +674,13 @@ def sample_template(self):
> >          -------
> >          SampleTemplate id
> >          """
> > -        return self._id
> > +        with TRN:
> > +            sql = """SELECT EXISTS(SELECT *
> > +                                   FROM qiita.study_sample
> > +                                   WHERE study_id = %s)"""
> > +            TRN.add(sql, [self.id])
> > +            exists = TRN.execute_fetchlast()
> > +        return self._id if exists else None
> 
> Good point, this has caused problems in the past. What about returning -1? 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiita/pull/1481/files#r41207203